src/libs/OptionsListUtils.ts

```diff
@@ -1978,25 +2027,57 @@ function getValidOptions(
             };
         }
 
-        personalDetailsOptions = getValidPersonalDetailOptions(options.personalDetails, {
-            loginsToExclude: personalDetailLoginsToExclude,
-            shouldBoldTitleByDefault,
-            includeDomainEmail,
-            currentUserRef,
-        });
+        const searchTerms = deburr(searchString ?? '')
+            .toLowerCase()
+            .split(' ')
+            .filter((term) => term.length > 0);
+        const filteringFunction = (personalDetail: OptionData) => {
+            if (
+                !personalDetail?.login ||
+                !personalDetail.accountID ||
+                !!personalDetail?.isOptimisticPersonalDetail ||
+                (!includeDomainEmail && Str.isDomainEmail(personalDetail.login)) ||
+                // Exclude the setup specialist from the list of personal details as it's a fallback if guide is not assigned
+                personalDetail?.login === CONST.SETUP_SPECIALIST_LOGIN
+            ) {
+                return false;
+            }
+            if (personalDetailLoginsToExclude[personalDetail.login]) {
+                return false;
+            }
+            const searchText = deburr(`${personalDetail.text ?? ''} ${personalDetail.login ?? ''}`.toLocaleLowerCase());
+
+            return searchTerms.length > 0 ? searchTerms.every((term) => searchText.includes(term)) : true;
+        };
+
+        personalDetailsOptions = optionsOrderBy(options.personalDetails, personalDetailsComparator, maxElements, filteringFunction, true);
+
+        for (let i = 0; i < personalDetailsOptions.length; i++) {
+            const personalDetail = personalDetailsOptions.at(i);
+            if (!personalDetail) {
+                continue;
+            }
+            if (!!currentUserLogin && personalDetail?.login === currentUserLogin) {
+                currentUserRef.current = personalDetail;
+            }
+            personalDetail.isBold = shouldBoldTitleByDefault;
+        }
     }
 
     if (excludeHiddenThreads) {
         recentReportOptions = recentReportOptions.filter((option) => !option.isThread || option.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN);
     }
 
+    let userToInvite: OptionData | null = null;
```

Comment:
NAB: `filterUserToInvite` returns `SearchOptionData | null` but it's being assigned to `OptionData | null` which creates a type mismatch.

```suggestion
    let userToInvite: SearchOptionData | null = null;
```